src/plugin/context.rs

```diff
@@ -883,3 +947,609 @@ impl RapierContext {
+    /// returns the `Entity` containing that rigid-body.
+    pub fn collider_parent(&self, entity: Entity) -> Option<Entity> {
+        for (_, world) in self.worlds.iter() {
+            if let Some(entity) = self.get_collider_parent_from_world(entity, world) {
+                return Some(entity);
+            }
+        }
+
+        None
+    }
+
+    /// If the collider attached to `entity` is attached to a rigid-body, this
+    /// returns the `Entity` containing that rigid-body.
+    ///
+    /// If the world does not exist, this returns None
+    pub fn collider_parent_for_world(&self, entity: Entity, world_id: WorldId) -> Option<Entity> {
+        if let Some(world) = self.worlds.get(&world_id) {
+            self.get_collider_parent_from_world(entity, world)
+        } else {
+            None
+        }
+    }
+
+    /// Retrieve the Bevy entity the given Rapier collider (identified by its handle) is attached.
+    pub fn collider_entity(&self, handle: ColliderHandle) -> Option<Entity> {
+        for (_, world) in self.worlds.iter() {
+            let entity = RapierWorld::collider_entity_with_set(&world.colliders, handle);
+            if entity.is_some() {
+                return entity;
+            }
+        }
+
+        None
+    }
+
+    /// Retrieve the Bevy entity the given Rapier rigid-body (identified by its handle) is attached.
+    pub fn rigid_body_entity(&self, handle: RigidBodyHandle) -> Option<Entity> {
+        for (_, world) in self.worlds.iter() {
+            let entity = world.rigid_body_entity(handle);
+            if entity.is_some() {
+                return entity;
+            }
+        }
+
+        None
+    }
+
+    /// Retrieve the Bevy entity the given Rapier rigid-body (identified by its handle) is attached.
+    ///
+    /// Returns None if this world does not exist
+    pub fn rigid_body_entity_in_world(
+        &self,
+        handle: RigidBodyHandle,
+        world_id: WorldId,
+    ) -> Option<Entity> {
+        self.worlds
+            .get(&world_id)
+            .map(|world| world.rigid_body_entity(handle))
+            .unwrap_or(None)
+    }
+
+    /// Advance the simulation, based on the given timestep mode.
+    #[allow(clippy::too_many_arguments)]
+    pub fn step_simulation(
+        mut self,
+        timestep_mode: TimestepMode,
+        mut events: Option<(EventWriter<CollisionEvent>, EventWriter<ContactForceEvent>)>,
+        hooks: &dyn PhysicsHooks,
+        time: &Time,
+        sim_to_render_time: &mut SimulationToRenderTime,
+        mut interpolation_query: Option<
+            &mut Query<(&RapierRigidBodyHandle, &mut TransformInterpolation)>,
+        >,
+    ) {
+        for (world_id, world) in self.worlds.iter_mut() {
+            world.step_simulation(
+                *world_id,
+                timestep_mode,
+                events.is_some(),
+                hooks,
+                time,
+                sim_to_render_time,
```

Comment:
I'm not sure sharing `SimulationToRenderTime` between all worlds will still yield useful results. I think we'd want it to be specific to each world.